stardis/radiation_field/radiation_field_solvers/base.py

```diff
@@ -117,78 +115,100 @@ def single_theta_trace_parallel(
     # Need to calculate a mean opacity for the traversal between points. Linearly interporlating. Van Noort paper suggests interpolating
     # alphas in log space. We could have a choice for interpolation scheme here.
     mean_alphas = np.exp((np.log(alphas[1:]) + np.log(alphas[:-1])) * 0.5)
-    taus = (
-        mean_alphas * geometry_dist_to_next_depth_point.reshape(-1, 1) / np.cos(theta)
-    )
-    no_of_depth_gaps = len(geometry_dist_to_next_depth_point)
+
+    taus = np.zeros_like(mean_alphas, dtype=np.float64)
+    for nu_index in numba.prange(taus.shape[1]):
+        for gap_index in range(taus.shape[0]):
+            taus[gap_index, nu_index] = (
+                mean_alphas[gap_index, nu_index]
+                * ray_dist_to_next_depth_point[gap_index]
+            )
+
+    no_of_depth_gaps = len(ray_dist_to_next_depth_point)
 
     source = source_function(tracing_nus, temps)
     I_nu_theta = np.zeros((no_of_depth_gaps + 1, len(tracing_nus)))
-    I_nu_theta[0] = source[
-        0
-    ]  # the innermost depth point is approximated as a blackbody
+    # the innermost depth point is approximated as a blackbody - changed to 0 for spherical geometry case. This should be fine if models are optically thick.
 
     w0, w1, w2 = calc_weights_parallel(taus)
 
     for nu_index in numba.prange(len(tracing_nus)):
         for gap_index in range(no_of_depth_gaps - 1):
             # Start by solving all the weights and prefactors except the last jump which would go out of bounds
-            second_term = (
-                w1[gap_index, nu_index]
-                * (
-                    (source[gap_index + 1, nu_index] - source[gap_index + 2, nu_index])
-                    * (taus[gap_index, nu_index] / taus[gap_index + 1, nu_index])
-                    - (source[gap_index + 1, nu_index] - source[gap_index, nu_index])
-                    * (taus[gap_index + 1, nu_index] / taus[gap_index, nu_index])
+            if taus[gap_index, nu_index] == 0:
+                I_nu_theta[gap_index + 1, nu_index] = I_nu_theta[
+                    gap_index, nu_index
+                ]  # If no optical depth, no change in intensity
+            else:
+                second_term = (
+                    w1[gap_index, nu_index]
+                    * (
+                        (
+                            source[gap_index + 1, nu_index]
```

Comment:
Some explanatory comments regarding the indices here would be nice. Or an equation reference.

Reply:
The equation reference is in the docstring. I think I didn't actually end up changing the raytracing equation other than not attentuating the ray for 0 optical depth. The rest of the equation just looks so different because black decided to format it differently this time for some reason I don't quite understand.

Reply:
On the note of indices though, do you think nu_index and gap_index aren't descriptive enough? I thought the names would be enough because they keep track of the gap in depth points you're moving past, or the frequency you're working on. But I've also been staring at this equation for months so I might be blind to it being unclear.

Reply:
Sorry, must have missed it in the docstring. The black formatting probably doesn't help either. The index names seem okay, though "gap" is not entirely clear to me. So you could either add the above as a descriptive comment, or find an alternative name.